### ISSUE TITLE
Return to first point

### ISSUE THREAD

**Issue by squishey123** [Courseplay/courseplay#7450]
*Mod version 6.4.0.0 from the mod hub*
To this date, I have only seen the return to first point option work once and it was about 1 tractor length of space between the start and end of the course (Using the field course generator), I have read through all the information within the manual a few times and could not find any info on any built-in limitations to the return to first point option.

Use cases I would like to use the option in but have not been able to:
1. Multiple tools being used and the final row of the course is not evenly spaced so implements get crowded towards the end and get stuck. If they were able to path to the beginning of the course and park next to each other I would not have to finish the final row manually.
2. Working on large fields where the start point is much closer to the parking spaces than the endpoint, saving driving time when I take over manual control.

On a side note, the CP driver does not seem to like to turn the engines off regardless of return to first point option set when work is done.

*Edit: After more experimenting, the feature seems to be more reliable when using "activated and release CP driver" with a re-supply route appended to the end. Would still love some documentation on the return to first point feature regarding limitations*

**Comment by rohne83** [comment]
I experience the 2nd use case very often with combines. While I can send home normal field workers (sprayers, cultivators, etc) directly using AD, I need the combines to go directly back to the start point where the header trailers are. No matter which setting I used for the return to first point, it never worked as I hoped with the latest CP versions (but if I remember correctly I had a few situations already when it worked indeed), which on large fields with multiple combines can be a bit frustrating ;)

**Comment by Tensuko** [comment]
logfile ?!

>     1. Multiple tools being used and the final row of the course is not evenly spaced so implements get crowded towards the end and get stuck. If they were able to path to the beginning of the course and park next to each other I would not have to finish the final row manually.

Screenshot of this situation pls.
Is it at the last Wp of the course or at the startpoint.


>     2\. Working on large fields where the start point is much closer to the parking spaces than the endpoint, saving driving time when I take over manual control.

Should be no Problem...


> Would still love some documentation on the return to first point feature regarding limitations

There are no limitations.
It might depend on the tool that you use.
I am not sure how it is when you have a collecting wagon and it is not full.

**Comment by squishey123** [comment]
Jumped into ravenport as a test world and came back with interesting results. Test was 3x of the 8m subsoilers from base game doing a basic spiral course with 1 headland, convoy mode enabled, all 3 vehicles set to Activated for return to first point.

Once vehicle 1 reached the end point he folded his tool and sat at the endpoint, the other 2 vehicles stopped at their normal 100m follow distance(As seen in Example1). After sitting idle for a few minutes I selected Go now to try to force him to route back to the beginning since he was obviously not going to do it of his own will. Driver 1 decided rather than pathing to the beginning he wanted to go on an adventure (Seen in Example 2) so I disabled convoy for him and the other 2 resumed driving. Once driver 2 reached the end he folded his tool and successfully pathed to the beginning but even once reaching the final waypoint driver 3 thought he was still 99m away and would not drive until convoy was disabled for driver 2. Like driver 2, driver 3 made it back to the beginning without any intervention. I'd assume if I had chosen activated and release CP driver then the convoy would not have to be disabled like I did however that doesn't change the fact that driver 1 refused to leave the final waypoint and went on an odd adventure.

When using tools like a combine the driver will call for an unload once reaching the final waypoint but even after being completely emptied he will sit at the final waypoint with his pipe out, regardless of wether the trailer is near him still or not.

Example1 (EX1), Example2 (EX2) and log file attached below.
Thanks!


![FS19_EX1](https://user-images.githubusercontent.com/16481004/131262301-60c7846e-9f7d-4f4b-b4a8-729410152bda.png)
![FS19_EX2](https://user-images.githubusercontent.com/16481004/131262302-00063c51-f9d6-45eb-956d-b4d6f283c99a.png)
[log.txt](https://github.com/Courseplay/courseplay/files/7073071/log.txt)
